OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -119,17 +123,7 @@ public async Task<WorkshopDto> Create(WorkshopCreateUpdateDto dto)
         workshop.Provider = await providerRepository.GetById(workshop.ProviderId).ConfigureAwait(false);
         workshop.ProviderOwnership = workshop.Provider.Ownership;
 
-        var tags = new List<TagDto>();
-
-        foreach (var tagId in dto.TagIds)
-        {
-            var tag = await tagService.GetById(tagId);
-            if (tag != null)
-            {
-                var tagDto = mapper.Map<TagDto>(tag);
-                tags.Add(tagDto);
-            }
-        }
+        workshop.Tags = (await tagRepository.GetAll()).Where(tag => dto.TagIds.Contains(tag.Id)).ToList();
```

Comment:
i do like this one query approach.

But you use both `ITagService` and `IEntityRepository<long, Tag>`. It will take the same amount of time to create this method in the service and use it instead.